provider-ci/internal/pkg/templates/base/Makefile

```
@@ -257,7 +258,7 @@ provider: bin/$(PROVIDER)
 # To create a release ready binary, you should use `make provider`.
 provider_no_deps:
 	$(call build_provider_cmd,$(shell go env GOOS),$(shell go env GOARCH),$(WORKING_DIR)/bin/$(PROVIDER))
-bin/$(PROVIDER): .make/schema
+bin/$(PROVIDER): .make/schema $(shell go run $(HELP_MAKE_GO) provider/cmd/$(PROVIDER))
```

It might be cleaner to simplify to:
```suggestion
bin/$(PROVIDER): .make/schema $(shell $(HELP_MAKE_GO) provider/cmd/$(PROVIDER))
```

Then we would change `HELP_MAKE_GO`:

```patch
-HELP_MAKE_GO := github.com/iwahbe/helpmakego@v0.1.0
+HELP_MAKE_GO := go run github.com/iwahbe/helpmakego@v0.1.0
```